### PR TITLE
feat(subl): add options

### PR DESCRIPTION
**Pull request by SeparateRecords**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This PR adds options to `subl`

### CODE REVIEW

File: src/subl.ts [hunk 1]
```diff
@@ -1,10 +1,83 @@
...
+        out.push(suggestion);
+        continue;
+      }
+      if (/^\.sublime-(project|workspace)$/.test(suggestion.name)) {
+        suggestion.priority = 75;
+        out.push(suggestion);
+      }
+    }
+    return out;
+  },
```
Comment (QuiiBz) [inline src/subl.ts:16]: I would prefer a cleaner solution, something like:
```suggestion
  filterTemplateSuggestions: (suggestions) => suggestions.map(suggestion => {
    if (suggestion.type === "folder") {
      return suggestion;
    }

    return {
      ...suggestion,
      priority: /^\.sublime-(project|workspace)$/.test(suggestion.name)
        ? 75
        : suggestion.priority
    }
  })
```

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for this PR!

File: src/subl.ts [hunk 2]
```diff
@@ -1,10 +1,83 @@
...
+        name: "command",
+      },
+    },
+    {
+      name: ["-n", "--new-window"],
+      description: "Open a new window",
+    },
+    {
+      name: ["-a", "--add"],
+      description: "Add folders to the current window",
```
[the remark this answers is not in the document]
Comment (SeparateRecords) [reply src/subl.ts:50]: Not by itself, it's just a modifier. When it's present, ST will add the files/dirs to the current window

File: src/subl.ts [hunk 3]
```diff
@@ -1,10 +1,83 @@
...
+      args: {
+        name: "project",
+        generators: generateProjects,
+      },
+    },
+    {
+      name: "--command",
+      description: "Run the given command",
+      args: {
+        name: "command",
```
[the remark this answers is not in the document]
Comment (SeparateRecords) [reply src/subl.ts:41]: It's not the kind of command Fig provides completions for, though - it's one of [these](https://docs.sublimetext.io/reference/commands.html#about-paths-in-command-arguments). Should it still be set?

File: src/subl.ts (hunk 1 above)
Comment (SeparateRecords) [reply src/subl.ts:16]: I think the actual problem here was my lack of comments explaining the intention :)
This loop also removes the files that _don't_ match the regex, so it's a combined filter/map. The equivalent would be this, which is slower. Debatably clearer, but comments can fix that in the imperative version.

```typescript
const generateProjects: Fig.Generator = {
  template: "filepaths",
  filterTemplateSuggestions: (suggestions) =>
    suggestions
      .filter((suggestion) => {
        if (suggestion.type === "folder") {
          return true;
        }
        return /^\.sublime-(project|workspace)$/.test(suggestion.name);
      })
      .map((suggestion) => {
        if (suggestion.type === "folder") {
          return suggestion;
        }
        return { ...suggestion, priority: 75 };
      }),
};
```

Lmk if you'd rather go ahead with this version!

File: src/subl.ts [hunk 4]
```diff
@@ -1,10 +1,83 @@
...
+      },
+    },
+    {
+      name: ["-n", "--new-window"],
+      description: "Open a new window",
+    },
+    {
+      name: ["-a", "--add"],
+      description: "Add folders to the current window",
+    },
```
Comment (SeparateRecords) [inline src/subl.ts:51]: @QuiiBz -n and -a aren't mutually exclusive, but when -n is present -a is silently ignored. Should these be marked as `exclusiveOn`?

File: src/subl.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (QuiiBz) [reply src/subl.ts:41]: Thanks for the clarification! So no, no need to set `isCommand`, you're 100% right!

File: src/subl.ts (hunk 1 above)
Comment (QuiiBz) [reply src/subl.ts:16]: That would be awesome if you could add some comments!

File: src/subl.ts (hunk 4 above)
Comment (QuiiBz) [reply src/subl.ts:51]: Totally! You can add `-a` to `exclusiveOn` for the `"-n", "--new-window"` option

**Review by mschrage — APPROVED** [review]
🙌

**Pull request merged**